RF_PBMC.R

```diff
@@ -0,0 +1,119 @@
+	print(jackstraw)
+	elbow <- PCElbowPlot(object = seuratObj)
+	print(elbow)
+	seuratObj <- FindClusters(object = seuratObj, reduction.type = "pca", dims.use = 1:10, resolution = 0.6, print.output = 0, save.SNN = TRUE)
+	PrintFindClustersParams(object = seuratObj)
+	seuratObj <- RunTSNE(object = seuratObj, dims.use = 1:10, do.fast = TRUE)
+	TSNEPlot(object = seuratObj)
+	# find all markers of cluster 1
+	cluster1.markers <- FindMarkers(object = seuratObj, ident.1 = 1, min.pct = 0.25)
+	print(x = head(x = cluster1.markers, n = 5))
+	# find all markers distinguishing cluster 5 from clusters 0 and 3
+	cluster5.markers <- FindMarkers(object = seuratObj, ident.1 = 5, ident.2 = c(0, 3), min.pct = 0.25)
+	print(x = head(x = cluster5.markers, n = 5))
+	# find markers for every cluster compared to all remaining cells, report
+	# only the positive ones
+	seuratObj.markers <- FindAllMarkers(object = seuratObj, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
+	seuratObj.markers %>% group_by(cluster) %>% top_n(2, avg_logFC)
+	cluster1.markers <- FindMarkers(object = seuratObj, ident.1 = 0, thresh.use = 0.25, test.use = "roc", only.pos = TRUE)
+	top10 <- seuratObj.markers %>% group_by(cluster) %>% top_n(10, avg_logFC)
+	return(seuratObj)
+}
+
+# generate random forest classifier 
+# save in tsv file - gene names for top 100 factors from the classification model in order of importance
+classify_data <- function(seuratObj){
+	classifier_input_x = t(seuratObj@data)
+	print(classifier_input_x)
+	classifier_input_y = as.factor(as.numeric(seuratObj@ident))
+	trained_classifier = randomForest(x = as.matrix(classifier_input_x) , y=factor(classifier_input_y) , importance = TRUE )
+	print(trained_classifier) 
+	ranked_list = trained_classifier$importance[order(-trained_classifier$importance[,'MeanDecreaseAccuracy']),]
+	dput(rownames(ranked_list)[1:300]) 
+	FINAL_LIST = rownames(ranked_list)[1:100]
+
+	write.table(FINAL_LIST, file='FINAL_LIST.tsv', quote=FALSE, sep='\t', row.names = FALSE , col.names = TRUE )
+}
+
+
+
+# Command line arguments
+pargs <- optparse::OptionParser(usage=paste("%prog [options]",
+                                            "--input_dir directory"))
+
+
+# input (string) path to Cell Ranger output directory
+pargs <- optparse::add_option(pargs, c("--input_dir"),
+                              type="character",
+                              action="store",
+                              dest="input_dir",
+                              metavar="Input_Directory",
+                              help=paste("Input directory for analysis.",
+                                         "[Default %default][REQUIRED]"))
+
+
+
+args_parsed <- optparse::parse_args(pargs)
+
+
+
+# Make sure the input directory exists
+if(!file.exists(args_parsed$input_dir[1])){
+    error_message <- paste0("Provided input directory, ",
+							args_parsed$input_dir[1], ",  does not exist.",
+    						"Please check your input and try again")
```

Comment:
Make spacing consistent.in 103-105. If you are using tabs please do not use tabs in any code. Use other spacing.

Reply:
will clean up tabs and use spaces instead